rust/operator-binary/src/crd.rs

```diff
@@ -131,6 +136,51 @@ impl AutoTlsCa {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub enum TlsKeyGeneration {
+    Rsa {
+        #[schemars(schema_with = "tls_key_length_schema")]
+        length: u32,
+    },
+}
+
+impl TlsKeyGeneration {
+    fn default_tls_key_generation() -> Self {
+        Self::Rsa { length: 4096 }
```

Comment:
Reading on https://github.com/ssllabs/research/wiki/SSL-and-TLS-Deployment-Best-Practices#31-avoid-too-much-security I think we should actually default to 2048, as secret-op only has 100m CPU

Reply:
Closing as agreed in daily